### PR TITLE
Components: Introduce Contextual Upgrade Trigger

### PR DESCRIPTION
**Pull request by renatoagds**
Introduce a new component, which will be used to push users to a higher plan or paid plan, if the current is free.

#### Changes proposed in this Pull Request:

* Add `ContextualUpgradeTrigger` component.

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

PbJqbW-5k-p2

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Start `Storybook`
* Navigate to `JS Packages/Components/Contextual Upgrade Trigger`.
* Check if works and match the proposed specs.

#### DEMO

image

### CODE REVIEW

**Review by oskosk — APPROVED** [review]
This works great!

I saw some strange behaviour happening if the cta is long. The longer the text is, the small the arrow gets. But I'd say to merge this one PR and address later if needed

image


image

**Comment by renatoagds** [comment]
r248117-wpcom

**Comment by keoshi** [comment]
Thanks for doing this!

> The longer the text is, the small the arrow gets.

It seems like a `min-width: 32px` on the SVG solves that issue. You can also check out this Codepen for a different way of how I structured it: https://codepen.io/keoshi/pen/ZEegmwL

**Comment by renatoagds** [comment]
> It seems like a min-width: 32px

Yes! I'll do that in a follow-up PR

File: projects/js-packages/components/components/contextual-upgrade-trigger/types.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export type CutBaseProps = {
+	description: string;
+	cta: string;
+	onClick: () => void;
```
Comment (manzoorwanijk) [inline projects/js-packages/components/components/contextual-upgrade-trigger/types.ts:4]: There is an in-built `VoidFunction` for this.

```suggestion
	onClick: VoidFunction;
```

**Pull request merged**